textalyze.js

```diff
@@ -54,15 +56,27 @@ module.exports = { itemCounts, arrayFrom, sanitize };
 //
 // running the app
 //
-if (require.main === module) {
-  const string = 'Hello World';
+function main(args) {
```

Comment:
I really like this!  It makes `main` potentially testable.

Since main requires one argument and maybe cares about other arguments later, I'd do:

```javascript
function main(path, ...args) {
  ...
}
```

Or even just `function main(path) { ... }`.

That said, there's something to having the first argument be `args` since it's more in line with other languages that require a `main` function (C, C++, Java, etc.).